src/components/alert/alert.tsx

```diff
@@ -144,9 +148,13 @@ export class Alert implements OpenCloseComponent {
     this.requestedIcon = setRequestedIcon(StatusIcons, this.icon, this.color);
   }
 
+  componentDidLoad(): void {
+    connectOpenCloseComponent(this);
```

Comment:
maybe instead of calling this on `componentDidLoad()` we can call it in `setTransitionEl()` after the element has been set? That way we can ensure the transitionEl exists when this is called.

Reply:
This applies to all components using this interface.

Reply:
Seconded. 

Additionally, `componentDidLoad` runs only once in a component's lifecycle, so if you were to move the component to another element, `disconnectedCallback` would remove the listener and it would no longer be reattached. I think the same could happen if we only set it in `setTransitionEl` (e.g., util is wired up on render, element is removed then reattached but does not re-render). We could additionally introduce a util solely for updating the reference. I suggest this to avoid confusion on having the set up util be called on both `connectedCallback` and `ref` hooks.

Sidebar: We've been using `(dis)connect<Util>` for all of these component utils to guide devs on where to use it. Maybe it's time to change the naming convention to something more generic (e.g., `setUp`/`tearDown`) and rely more on the util doc?

Reply:
> We could additionally introduce a util solely for updating the reference. I suggest this to avoid confusion on having the set up util be called on both connectedCallback and ref hooks.

Scrap this. Synced up w/ @Elijbet, and there really wouldn't be much difference between the connect method and the ref updating util. Renaming the utils to move away from `connect`/`disconnect` + util doc should help avoid confusion. To recap: the connect util should be called in `setTransitionEl` and `connectedCallback` to guarantee the node is available.